plugins/org.eclipse.mat.parser/src/org/eclipse/mat/parser/internal/snapshot/IObjectMarker.java

```diff
@@ -0,0 +1,18 @@
+package org.eclipse.mat.parser.internal.snapshot;
+
+import org.eclipse.mat.SnapshotException;
+import org.eclipse.mat.snapshot.ExcludedReferencesDescriptor;
+import org.eclipse.mat.snapshot.ISnapshot;
+import org.eclipse.mat.util.IProgressListener;
+
+public interface IObjectMarker
+{
+
+    int markSingleThreaded();
+
+    void markMultiThreaded(int threads) throws InterruptedException;
```

Comment:
Does this become public API? If so, maybe mark Single vs Multi should be part of the ObjectMarkerFactory?

Reply:
No, since it's in the `internal` package, it doesn't become public API. Only these packages are exported: https://github.com/eclipse-mat/mat/blob/master/plugins/org.eclipse.mat.parser/META-INF/MANIFEST.MF#L13-L16